server/src/test/java/org/elasticsearch/search/lookup/SourceFilterTests.java

```diff
@@ -112,4 +112,48 @@ public Source filter(SourceFilter sourceFilter) {
 
     }
 
+    // Verification for issue #109668
+    public void testIncludeParentAndExcludeChildEmptyArray() {
+        Source fromMap = Source.fromMap(Map.of("myArray", List.of()), XContentType.JSON);
+        Source filteredMap = fromMap.filter(new SourceFilter(new String[] { "myArray" }, new String[] { "myArray.myField" }));
+        assertEquals(filteredMap.source(), Map.of("myArray", List.of()));
+        Source fromBytes = Source.fromBytes(new BytesArray("{\"myArray\": []}"), XContentType.JSON);
+        Source filteredBytes = fromBytes.filter(new SourceFilter(new String[] { "myArray" }, new String[] { "myArray.myField" }));
+        assertEquals(filteredBytes.source(), Map.of("myArray", List.of()));
+    }
+
+    public void testIncludeParentAndExcludeChildEmptyObject() {
+        Source fromMap = Source.fromMap(Map.of("myObject", Map.of()), XContentType.JSON);
+        Source filteredMap = fromMap.filter(new SourceFilter(new String[] { "myObject" }, new String[] { "myObject.myField" }));
+        assertEquals(filteredMap.source(), Map.of("myObject", Map.of()));
+        Source fromBytes = Source.fromBytes(new BytesArray("{\"myObject\": {}}"), XContentType.JSON);
+        Source filteredBytes = fromBytes.filter(new SourceFilter(new String[] { "myObject" }, new String[] { "myObject.myField" }));
+        assertEquals(filteredBytes.source(), Map.of("myObject", Map.of()));
+    }
+
+    public void testIncludeParentAndExcludeChildSubFieldsArrays() {
```

Comment:
Note to self: This test passes even without the fix.

Reply:
I think @mccheah intention was to test map based filtering for this behaviour, which did work before the fix. Can you please confirm @mccheah ?

Reply:
Not exactly - though this does inform me I didn't write this in the most intuitive way, and if I were to do this again I'd adjust it -

There's two test cases embedded in this. Assuming no patch: The first does serde from Map, which passes. The second does serde with bytes, which fails.

What I would self-recommend having done instead was creating separate test cases for each of map and byte serde.

Reply:
Oh wait never mind - In the tests that pass without the fix, I was making sure I didn't introduce regression in more "standard" cases via the patch I had written. Sorry, got mixed up in my mind =)

Reply:
That was my assumption as well 👍